vignettes/REDCapTidieR.Rmd

```diff
@@ -10,21 +10,37 @@ vignette: >
 ---
 
 ```{r, include = FALSE}
-NOT_CRAN <- identical(tolower(Sys.getenv("NOT_CRAN")), "true")
 knitr::opts_chunk$set(
   collapse = TRUE,
-  comment = "#>",
-  purl = NOT_CRAN,
-  eval = NOT_CRAN
+  comment = "#>"
 )
 ```
 
 ```{r, include=FALSE}
+library(httptest)
+start_vignette("REDCapTidieR")
+
 library(dplyr)
+# If creating mocks make sure REDCapTidieR is attached prior to any calls that
+# need to be mocked. Otherwise our custom redactor won't be found when knitting
+# TLDR: Don't move this library call
 library(REDCapTidieR)
 
-token <- Sys.getenv("SUPERHEROES_REDCAP_API")
-redcap_uri <- Sys.getenv("REDCAP_URI")
+# If we're creating mocks we need to use the real tokens. Otherwise, use fakes.
+
+# Check whether we're mocking
+# start_vignette() starts mocking if the mock directory doesn't exist
+# Checking this way is slightly hacky but httptest doesn't seem to expose
+# anything that would allow an "official" check
+mocking <- !dir.exists("REDCapTidieR")
+
+if (mocking) {
```

Comment:
I'm not sure this works. What if you've added a new API call to the Rmd? In this case you'd have the fake creds set but nothing cached. 

I believe what you want to do here is use the env vars if they are available.

Reply:
I think I follow but the way `start_vignette()` works means you will either be creating new mocks for the entire vignette or using existing mocks for the entire vignette. The [recommended workflow](https://enpiar.com/r/httptest/articles/vignettes.html) for adding a new mock like the case you described is to delete the directory with mocks for the vignette and then knit it to regenerate all of the mocks.

If we supplied env vars when the vignette isn't creating mocks it would fail because it would be looking for mocks with the fake credentials.

Reply:
Ah interesting. I like that approach. Let me see if I can get it to work